tests/integration/examples.py

```diff
@@ -11,19 +12,29 @@
 from testing import run
 
 
+def svstat_stdout_check(service, state):
+    p = Popen(('svstat', 'playground/{}'.format(service)), stdout=PIPE, stderr=PIPE)
+    stdout, stderr = run(p)
+    assert stderr == ''
+    assert 'playground/{0}: {1}'.format(service, state) in stdout
+
+
 class DescribeDateExample(object):
 
     @fixture
     def service_name(self):
         yield 'date'
 
-    def it_does_start(self, in_example_dir):
+    def it_does_start(self, in_example_dir, request):
         assert not os.path.isfile('now.date')
-        p = Popen(('pgctl-2015', 'start', 'date'))
-        p.wait()
-        assert p.returncode == 0
+        check_call(('pgctl-2015', 'start', 'date'))
+
         assert os.path.isfile('now.date')
 
+        def fin():
+            check_call(('pgctl-2015', 'stop', 'date'))
+        request.addfinalizer(fin)
```

Comment:
generally you want to add your finalizer *before* the spots where you might throw.